lib/ruby_indexer/lib/ruby_indexer/entry.rb

```diff
@@ -40,6 +40,15 @@ class Namespace < Entry
 
       abstract!
 
+      sig { returns(SingletonClass) }
+      attr_reader :singleton_klass
+
+      sig { params(name: String, file_path: String, location: Prism::Location, comments: T::Array[String]).void }
+      def initialize(name, file_path, location, comments)
+        super
+        @singleton_klass = T.let(SingletonClass.new(name, self), SingletonClass)
```

Comment:
```suggestion
        @singleton_class = T.let(SingletonClass.new(name, self), SingletonClass)
```

As the other one is `attached_class`.

Reply:
We can't call this `@singleton_class` or else its accessor would be called `singleton_class` and that conflicts with the Ruby internal method. It also throws Sorbet off.